scene/resources/motion_blur_compositor_effect.cpp

```diff
@@ -0,0 +1,286 @@
+	overlay_shader_file->parse_versions_from_text(overlay_glsl);
+	Ref<RDShaderSPIRV> overlay_bytecode = overlay_shader_file->get_spirv();
+	overlay_shader = RD::get_singleton()->shader_create_from_spirv(overlay_bytecode->get_stages(), "Motion blur overlay");
+	overlay_pipeline = RD::get_singleton()->compute_pipeline_create(overlay_shader);
+}
+
+RD::Uniform MotionBlurCompositorEffect::get_image_uniform(RID p_image, int p_binding) {
+	RD::Uniform uniform;
+	uniform.uniform_type = RD::UNIFORM_TYPE_IMAGE;
+	uniform.binding = p_binding;
+	uniform.append_id(p_image);
+	return uniform;
+}
+
+RD::Uniform MotionBlurCompositorEffect::get_sampler_uniform(RID p_image, int p_binding) {
+	RD::Uniform uniform;
+	uniform.uniform_type = RD::UNIFORM_TYPE_SAMPLER_WITH_TEXTURE;
+	uniform.binding = p_binding;
+	uniform.append_id(linear_sampler);
+	uniform.append_id(p_image);
+	return uniform;
+}
+
+void MotionBlurCompositorEffect::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("_render_callback", "effect_callback_type", "render_data"), &MotionBlurCompositorEffect::_render_callback);
+
+	ClassDB::bind_method(D_METHOD("get_motion_blur_samples"), &MotionBlurCompositorEffect::get_motion_blur_samples);
+	ClassDB::bind_method(D_METHOD("set_motion_blur_samples", "samples"), &MotionBlurCompositorEffect::set_motion_blur_samples);
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "motion_blur_samples"), "set_motion_blur_samples", "get_motion_blur_samples");
+
+	ClassDB::bind_method(D_METHOD("get_motion_blur_intensity"), &MotionBlurCompositorEffect::get_motion_blur_intensity);
+	ClassDB::bind_method(D_METHOD("set_motion_blur_intensity", "intensity"), &MotionBlurCompositorEffect::set_motion_blur_intensity);
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "motion_blur_intensity", PROPERTY_HINT_RANGE, "0,0.5,or_greater"), "set_motion_blur_intensity", "get_motion_blur_intensity");
+
+	ClassDB::bind_method(D_METHOD("get_motion_blur_center_fade"), &MotionBlurCompositorEffect::get_motion_blur_center_fade);
+	ClassDB::bind_method(D_METHOD("set_motion_blur_center_fade", "center_fade"), &MotionBlurCompositorEffect::set_motion_blur_center_fade);
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "motion_blur_center_fade", PROPERTY_HINT_RANGE, "0,1,or_greater"), "set_motion_blur_center_fade", "get_motion_blur_center_fade");
+}
+
+void MotionBlurCompositorEffect::_render_callback(int p_effect_callback_type, const RenderDataRD *p_render_data) {
+	if (RD::get_singleton() && p_effect_callback_type != EFFECT_CALLBACK_TYPE_POST_TRANSPARENT) {
+		return;
+	}
+	Ref<RenderSceneBuffersRD> render_scene_buffers = p_render_data->get_render_scene_buffers();
+	RenderSceneData *render_scene_data = p_render_data->get_render_scene_data();
+	if (render_scene_buffers.is_null() || !render_scene_data) {
+		return;
+	}
+	Size2i render_size = render_scene_buffers->get_internal_size();
+	if (render_size.x == 0 || render_size.y == 0) {
+		return;
+	}
+
+	if (render_scene_buffers->has_texture(context, texture)) {
```

Comment:
```suggestion
	const StringName context = SNAME("MotionBlur");
	const StringName texture = SNAME("texture");
	if (render_scene_buffers->has_texture(context, texture)) {
```
I'd say, this avoids converting each time and keeps them local